ocaml/xapi/xapi_pool.ml

```diff
@@ -3609,3 +3609,55 @@ let set_https_only ~__context ~self:_ ~value =
         )
         (Db.Host.get_all ~__context)
   )
+
+let configure_update_sync ~__context ~self ~update_sync_frequency
+    ~update_sync_day ~update_sync_hour =
+  ( match update_sync_frequency with
+  | `daily when update_sync_day <> 1L ->
```

Comment:
I'm not sure if this is correct that no variables in `when` come from matched pattern.
Following let me feel easy. Would appreciate if anyone could have an elaboration here. 
```
match (update_sync_frequency, update_sync_day) with
| `daily, d when d <> 1L
......
```

Reply:
And would following look better?
```
let day_of_month = function
  | i when i >=1 and i <= 31 ->
      Some i
  | _ ->
      None
in
let hour_of_day = function
  | i when i >=0 and i <= 23 ->
      Some i
  | _ ->
      None
in
let day_of_week = function
  | i when i >= 1 and i <= 7 ->
      Some i
  | _ ->
      None
in
match (frequency,
       day_of_month(update_sync_day),
       day_of_week(update_sync_day),
       hour_of_day(update_sync_hour)) with
| `daily, _, _, Some _
| `weekly, _, Some _, Some _
| `monthly, Some _, _, Some _ ->
    ()
| _ ->
    raise ...
```

Reply:
I think having an explicit check for day_of_month, hour. day_of_week is good. But note that existing code allows for negative numbers, too. A `day_of_month = 31` is problematic. What happens in a month with 30 days? Is this synchronization triggered or not in that situation?  This needs to be clearly documented.